### PR TITLE
Implement url2args excmd

### PR DESCRIPTION
**Pull request by glacambre**
The documentation explains what it does/how to use it. I didn't change the bindings to use it by default because it might break in surprising ways.

### CODE REVIEW

File: src/excmds.ts [hunk 1]
```diff
@@ -868,6 +868,33 @@ export async function open_quiet(...urlarr: string[]) {
...
+ *  Note that this might break with search engines that redirect you to other pages/add GET parameters that do not exist in your searchurl.
+ */
+//#content
+export async function url2args() {
+    let url = document.location.href
+    let searchurls = await config.getAsync("searchurls")
+    let result = url
+
+    for (let engine in searchurls) {
+        let [beginning, end] = [...searchurls[engine].split("%s"), ""]
```
Comment (bovine3dom) [inline src/excmds.ts]: The %s isn't guaranteed to exist. Does this still work without it?

Comment (glacambre) [reply src/excmds.ts]: Yup, "%s" not being there is the reason why I added the empty string at the end of the array. I can add comments to explain this if you want me to.

**Comment by glacambre** [comment]
It'd probably be a good idea to URL-decode the returned string. I'll do that tomorrow.

**Comment by bovine3dom** [comment]
This doesn't work for me. Steps to reproduce on a fresh profile:

```
:bind O composite url2args | fillcmdline open<CR>
stest<CR>
O
```
Result: `open google  `. Expected: `open google test`.

**Comment by glacambre** [comment]
Thanks for noticing this!

**Pull request merged**